src/controller/cpp/Proto.cpp

```diff
@@ -0,0 +1,73 @@
+// Copyright 1996-2023 Cyberbotics Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#define WB_ALLOW_MIXING_C_AND_CPP_API
+#include <webots/supervisor.h>
+#include <webots/Proto.hpp>
+
+#include <stdio.h>
+#include <map>
+
+using namespace std;
+using namespace webots;
+
+static map<WbProtoRef, Proto *> protoMap;
+
+Proto *Proto::findProto(WbProtoRef ref) {
+  if (!ref)
+    return NULL;
+
+  map<WbProtoRef, Proto *>::iterator iter = protoMap.find(ref);
+  if (iter != protoMap.end())
+    return iter->second;
+
+  Proto *field = new Proto(ref);
```

Comment:
Seems like `proto` would be a better name that `field` here.

Reply:
Yes it would! Fixed!